File: Stack/java/GENstack.java
class node<T>
{
    public T data;
    public node<T> next;
}


class StackX<T>
{
    private node<T> first;
    private int iCount;


    public StackX()
    {
        first = null;
        iCount = 0;
    }

    void Push(T iNo)   // insert first
    {
        node<T> newn = new node<T>();

        newn.data = iNo;
        newn.next = null;

        if(first == null)
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;   
        }
        iCount++;
    }

    public T Pop()               // delete first
    {
        node<T> temp = first;
        if(first == null)
        {
            System.out.println("Stack is empty \n");
            return null;
        }
        else
        {
            first = first.next;
            iCount--;
            return temp.data;

        }
      
    }
    
    void Display()
    {
        node<T> temp = first;


        while(temp != null)
        {
            System.out.println(" | "+temp.data+" | ");
            temp = temp.next;
        }
    }

    public int Count ()
    {
        return iCount;
    }

    T  Peek()
    {
        node<T> temp = first;
        if(temp == null)
        {
            System.out.println("stack is empty");
            return null;
        }
        return temp.data;
    }

    boolean isempty()
    {
        boolean bflage = false;

        if(first == null)
        {
            bflage= true;
        }
        return bflage;
    }
}

class GENstack
{
    public static void main(String a[])
    {
        StackX<Integer> sobj = new StackX<>();
        int iRet = 0;
        boolean bRet = false;

        sobj.Push(121);
        sobj.Push(111);
        sobj.Push(101);
        sobj.Push(51);
        sobj.Push(21);
        sobj.Push(11);
        

        sobj.Display();

        iRet = sobj.Count();
        System.out.println("no of element in stack :"+iRet);

        iRet = sobj.Pop();
        System.out.println("poped element is  :"+iRet);

        sobj.Display();

        iRet = sobj.Count();
        System.out.println("no of element in stack :"+iRet);

        iRet = sobj.Peek ();
        System.out.println("top  element in stack  is :"+iRet);

        bRet = sobj.isempty();
        if(bRet == true)
        {
            System.out.println("satck is empty");
        }
        else
        {
            System.out.println("stack is not empty");
        }
        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
        
        StackX<Double> sobj1 = new StackX<>();
        double dRet = 0.0;

        sobj1.Push(121.09);
        sobj1.Push(111.56);
        sobj1.Push(101.56);
        sobj1.Push(51.87);
        sobj1.Push(21.89);
        sobj1.Push(11.79);
        

        sobj1.Display();

        iRet = sobj1.Count();
        System.out.println("no of element in stack :"+iRet);

        dRet = sobj1.Pop();
        System.out.println("poped element is  :"+dRet);

        sobj1.Display();

        iRet = sobj1.Count();
        System.out.println("no of element in stack :"+iRet);

        dRet = sobj1.Peek ();
        System.out.println("top  element in stack  is :"+dRet);

        bRet = sobj1.isempty();
        if(bRet == true)
        {
            System.out.println("satck is empty");
        }
        else
        {
            System.out.println("stack is not empty");
        }
        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
        
        
        StackX<String> sobj3 = new StackX<>();
        String str = null;

        sobj3.Push("anii");
        sobj3.Push("sagar");
        sobj3.Push("suresh");
        sobj3.Push("chandu");
        sobj3.Push("omkar");
        sobj3.Push("trisha");
        

        sobj3.Display();

        iRet = sobj3.Count();
        System.out.println("no of element in stack :"+iRet);

        str = sobj3.Pop();
        System.out.println("poped element is  :"+str);

        sobj3.Display();

        iRet = sobj3.Count();
        System.out.println("no of element in stack :"+iRet);

        str = sobj3.Peek ();
        System.out.println("top  element in stack  is :"+str);

        bRet = sobj3.isempty();
        if(bRet == true)
        {
            System.out.println("satck is empty");
        }
        else
        {
            System.out.println("stack is not empty");
        }
        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
    }
}

File: Stack/CPP/GENstack.cpp
#include<iostream>
using namespace std;

template<class T>

struct node
{
    T data ;
    struct node<T> * next;
};

template<class T>

class StackX
{
    private: struct node<T> * first;
    private: int iCount ;

    public: StackX()
    {
        this->first = NULL;
        this->iCount= 0;
    }
    
    void Push (T no)
    {
        struct node<T> *newn = new struct node<T>();

        newn->data = no;
        newn->next = NULL;

        if(first == NULL)
        {
            first = newn;
        }
        else
        {
            newn->next = first;
            first = newn;
        }
        iCount++;
    }

    T Pop()
    {
        struct node<T>*temp = first;
        T data = 0;

        if(first == NULL)
        {
            cout<<"stack is empty no element for pop \n";
            return NULL;
        }
        else
        {
            first = temp->next;
            data =  temp->data;
            iCount--;
            delete temp;
            return data;
        }

    }

    void Display()
    {
        struct node<T> *temp =first;

        while(temp != NULL)
        {
            cout<<" | "<<temp->data<<" | -"<<"\n";
            temp = temp->next;
        }
    }

    int Count()
    {
        return iCount;
    }


};

int main()
{
    StackX<int> * sobj = new StackX<int>();
    int i = 0,ivalue = 0,isize = 0,iret = 0;

    cout<<"enter no of element you want to enter :\n";
    cin>>isize;

    for(i = 1; i <= isize ;i++)
    {
        cin>>ivalue;
        sobj->Push(ivalue);
    }
   
    sobj->Display();

    iret = sobj->Count();
    cout<<"no of element in stack is :"<<iret<<"\n";

    iret = sobj->Pop();
    cout<<" element poped from  stack is :"<<iret<<"\n";

    sobj->Display();

    iret = sobj->Count();
    cout<<"no of element in stack is :"<<iret<<"\n";

    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    StackX<double> * sobj1 = new StackX<double>();
    double dvalue = 0.0,dret = 0.0;

    cout<<"enter no of element you want to enter :\n";
    cin>>isize;

    for(i = 1; i <= isize ;i++)
    {
        cin>>dvalue;
        sobj1->Push(dvalue);
    }
   
    sobj1->Display();

    iret = sobj1->Count();
    cout<<"no of element in stack is :"<<iret<<"\n";

    dret = sobj1->Pop();
    cout<<" element poped from  stack is :"<<dret<<"\n";

    sobj1->Display();

    iret = sobj1->Count();
    cout<<"no of element in stack is :"<<iret<<"\n";

    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

     StackX<float> * sobj2 = new StackX<float>();
    float fvalue = 0.0,fret = 0.0;

    cout<<"enter no of element you want to enter :\n";
    cin>>isize;

    for(i = 1; i <= isize ;i++)
    {
        cin>>fvalue;
        sobj2->Push(fvalue);
    }
   
    sobj2->Display();

    iret = sobj2->Count();
    cout<<"no of element in stack is :"<<iret<<"\n";

    fret = sobj2->Pop();
    cout<<" element poped from  stack is :"<<fret<<"\n";

    sobj2->Display();

    iret = sobj2->Count();
    cout<<"no of element in stack is :"<<iret<<"\n";

    return 0;
}